src/seedu/addressbook/Main.java

```diff
@@ -74,12 +76,25 @@ private void exit() {
         ui.showGoodbyeMessage();
         System.exit(0);
     }
+    
+    private void checkStorageFile() throws NoStorageFileException {
+    	File storageFile = new File(DEFAULT_STORAGE_FILEPATH);
+    	if(!storageFile.exists()) {
+    		throw new NoStorageFileException("Storage file cannot be found");
+    	}
+    }
 
     /** Reads the user command and executes it, until the user issues the exit command.  */
     private void runCommandLoopUntilExitCommand() {
         Command command;
         do {
             String userCommandText = ui.getUserCommand();
+            try {
+				checkStorageFile();
```

Comment:
This method may be too high-level to detect the exception. The method doesn't even know there's a file involved. Perhaps `StorageFile.save()` is a better place to detect it?